src/osa/utils/utils.py

```diff
@@ -285,3 +292,137 @@ def wait_for_daytime(start=8, end=18):
+    """
+    Calculate culmination angle for a given declination.
+
+    Parameters
+    ----------
+    dec: Quantity
+        declination coordinate in degrees
+
+    Returns
+    -------
+    Culmination angle in degrees
+    """
+    location = observatory_locations["cta_north"]
+    Lat = location.lat  # latitude of the LST1 site    
+    return abs(Lat - dec)
+
+
+def get_median_dec(datacheck_file: Path) -> u.Quantity:
+    """Get the median pointing declination of a given run from the run-wise datacheck file."""
+    with tables.open_file(datacheck_file) as dcf:
+        return np.median(dcf.root.dl1datacheck.cosmics.col('tel_dec'))*u.deg
+
+
+def convert_dec_string(dec_str: str) -> u.Quantity:
+    """Return the declination angle in degrees corresponding to a 
+    given string of the form "dec_XXXX" or "dec_min_XXXX"."""
+    
+    # Check if dec_str has a valid format
+    pattern = r'^dec_(\d{3,4})$|^dec_min_(\d{3,4})$'
+    if re.match(pattern, dec_str):
+        
+        # Split the string into parts
+        parts = dec_str.split('_')
+
+        # Extract the sign, degrees, and minutes
+        sign = 1 if 'min' not in parts else -1
+        degrees = int(parts[-1])
+
+        # Calculate the numerical value
+        dec_value = sign * (degrees / 100)
+
+        return dec_value*u.deg
+
+
+def get_declinations_dict(list1: list, list2: list) -> dict:
+    """Return a dictionary created from two given lists."""
+    corresponding_dict = {}
+    for index, element in enumerate(list2):
+        corresponding_dict[element] = list1[index]
+    return corresponding_dict
+        
+
+def get_nsb_dict(rf_models_dir: Path, rf_models_prefix: str) -> dict:
+    """Return a dictionary with the NSB level of the RF models and the path to each model."""
+    rf_models = sorted(rf_models_dir.glob(f"{rf_models_prefix}*"))
+    pattern = r"nsb_tuning_([\d.]+)"
+    nsb_dict = {
+        float(re.search(pattern, str(rf_model)).group(1)): rf_model
+        for rf_model in rf_models if re.search(pattern, str(rf_model))
+    }
+    return nsb_dict
+
+
+def get_mc_nsb_dir(run_id: int, rf_models_dir: Path) -> Path:
+    """
+    Return the path of the RF models directory with the NSB level 
+    closest to that of the data for a given run.
+    """
+    input_dir = Path(options.directory)
+    _, additional_nsb, _ = find_tailcuts(input_dir, run_id)
+
+    rf_models_prefix = cfg.get("lstchain", "mc_prod")
+    nsb_dict = get_nsb_dict(rf_models_dir, rf_models_prefix)
+    closest_nsb_value = min(nsb_dict.keys(), key=lambda x: abs(float(x) - additional_nsb))
+
+    return nsb_dict[closest_nsb_value]
+
+
+def get_RF_model(run_id: int) -> Path:
+    """Get the path of the RF model to be used in the DL2 production for a given run."""
+    datacheck_dir = options.directory / options.dl1_prod_id / "datacheck"
+    datacheck_file = datacheck_dir / f"datacheck_dl1_LST-1.Run{run_id:05d}.h5"
+
+    source_dec = get_median_dec(datacheck_file)
```

Comment:
this is no longer source declination but *telescope pointing* declination. I would rename these variables to avoid confusion in the future.